### PR TITLE
Migrate reports screen to Mobius - Prepare controller

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.clubhouse.io/simpledotorg/story/342/migrate-reports-screen-to-mobius

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/home/report/ReportsModel.kt [hunk 1]
```diff
@@ -4,4 +4,9 @@ import android.os.Parcelable
 import kotlinx.android.parcel.Parcelize
 
 @Parcelize
-class ReportsModel : Parcelable
+class ReportsModel : Parcelable {
+
+  companion object {
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/home/report/ReportsModel.kt:9]: Can you tell me why do we need this here?

File: app/src/main/java/org/simple/clinic/home/report/ReportsScreen.kt [hunk 2]
```diff
@@ -34,7 +42,7 @@ class ReportsScreen(context: Context, attrs: AttributeSet) : FrameLayout(context
 
     bindUiToController(
         ui = this,
-        events = screenCreates(),
+        events = events,
         controller = controller,
         screenDestroys = detaches().map { ScreenDestroyed() }
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/home/report/ReportsScreen.kt:66]: Can we extract `screenDestroys` to a shared stream, just like we did for events, please?

File: app/src/main/java/org/simple/clinic/home/report/ReportsEvent.kt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package org.simple.clinic.home.report
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/home/report/ReportsEvent.kt:1]: Can you create `ReportsUiActions` interface and extend in `ReportsUi` in no-op implementation ?

**Comment by msasikanth** [comment]
> Looks good to me. Can you update the CHANGELOG file, please?

There is already a CHANGELOG entry for the same day for the migration.

File: app/src/main/java/org/simple/clinic/home/report/ReportsModel.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/home/report/ReportsModel.kt:9]: We are using it when creating `MobiusTestFixture` in `ReportsScreenControllerTest `

File: app/src/main/java/org/simple/clinic/home/report/ReportsScreen.kt (hunk 2 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/home/report/ReportsScreen.kt:66]: We don't need `screenDestroys` as part of the shared stream. It's only required when binding the controller here.

File: app/src/main/java/org/simple/clinic/home/report/ReportsEvent.kt (hunk 3 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/home/report/ReportsEvent.kt:1]: There are no `UiActions ` in this screen, so we don't have to create it.

File: app/src/main/java/org/simple/clinic/home/report/ReportsScreen.kt (hunk 2 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/home/report/ReportsScreen.kt:66]: @vinaysshenoy can you confirm this once? If I am not using `ScreenDestroyed` in the controller, and it's only being used in `ReplayUntilScreenIsDestroyed`. Do I still need to make the `screenDestroys` as part of the shared stream? 🤔

Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/home/report/ReportsScreen.kt:66]: No longer necessary. We will only need it if screen destroyed events are specifically used for something else.

**Pull request merged**